tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,305 @@
+            }],
+            options: ["except-simple"]
+        },
+
+        // global `expect` does not count when used inside of a callback
+        {
+            code: "test('name', function() { maybe(function() { expect(1); ok(true) }); });",
+            errors: [{
+                message: exceptSimpleErrorMessage("expect"),
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert in outer test context and nested in a block
+        {
+            code: "test('name', function(assert) { assert.ok(true); if (true) { assert.ok(true); } });",
+            errors: [{
+                message: exceptSimpleErrorMessage("assert.expect"),
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Deeply nested
+        {
+            code: "test('name', function() { if (true) { if (true) { ok(true); } } });",
+            errors: [{
+                message: exceptSimpleErrorMessage("expect"),
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Sending assert to a function
+        {
+            code: [
+                "function myAssertion(a, assert, c) { assert.ok(true); }",
+                "test('name', function(assert) { myAssertion(null, assert, null); });"
+            ].join(returnAndIndent),
+            errors: [{
+                message: exceptSimpleErrorMessage("assert.expect"),
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // Sending assert to a function - renaming assert
+        {
+            code: [
+                "function myAssertion(a, localAssert, c) { localAssert.ok(true); }",
+                "test('name', function(myAssert) { myAssertion(null, myAssert, null); });"
+            ].join(returnAndIndent),
+            errors: [{
+                message: exceptSimpleErrorMessage("myAssert.expect"),
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // nested modules
+        {
+            code: [
+                "module('a', function() {",
+                "    test('a - test', function(assert) {",
+                "        if (false) {",
+                "            assert.ok(true, 'needs expect');",
+                "        }",
+                "    });",
+                "",
+                "    module('b', function() {",
+                "        test('b - test', function(assert) {",
+                "            assert.expect(1);",
+                "            assert.ok(true, 'has expect');",
+                "        });",
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            errors: [{
+                message: exceptSimpleErrorMessage("assert.expect"),
+                type: "CallExpression"
```

Comment:
Hmm, this came out to be line 4. This happens because the check happens when leaving the test. I'll adjust this so that it reports the currentTest node.